native/core/src/parquet/parquet_support.rs

```diff
@@ -596,7 +595,10 @@ fn cast_array(
     parquet_options: &SparkParquetOptions,
 ) -> DataFusionResult<ArrayRef> {
     use DataType::*;
-    let array = array_with_timezone(array, parquet_options.timezone.clone(), Some(to_type))?;
+    let array = match to_type {
+        Timestamp(_, None) => array, // array_with_timezone does not support to_type of NTZ.
```

Comment:
This will skip all the matches inside `array_with_timezone`
```
    match array.data_type() {
        DataType::Timestamp(_, None) => {
            ...
            match to_type {
                ...
                Some(DataType::Timestamp(_, None)) => {
                    timestamp_ntz_to_timestamp(array, timezone.as_str(), None)
                }
                ...
```
Is it intentional?

Reply:
Yeah, it shouldn't be called with that type:
https://github.com/apache/datafusion-comet/blob/f099e6e40aa18441c7882e5bffd9d6dfb10c6c19/native/spark-expr/src/utils.rs#L59

The condition that matches inside `array_with_timezone` was added by #1348 and we likely didn't understand the blast radius at the time.

Reply:
In fact maybe I'll remove that in this PR if nothing breaks.

Reply:
Or what about updating the logic in `array_with_timezone` rather than fixing at this location?

Reply:
Yep I removed that case in utils.rs.